core/src/main/java/org/elasticsearch/action/support/single/instance/TransportInstanceSingleOperationAction.java

```diff
@@ -169,11 +173,11 @@ protected boolean doStart() {
 
             if (!shard.active()) {
                 retry(null);
-                return false;
+                return;
             }
 
             if (!operationStarted.compareAndSet(false, true)) {
```

Comment:
I don't think we need to protected ourselves here against violation of transport service semantics (just one callback).